staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -624,8 +633,38 @@ func DumpReaderToFile(reader io.Reader, filename string) error {
 	return nil
 }
 
-func GetServerSideApplyFlag(cmd *cobra.Command) bool {
-	return GetFlagBool(cmd, "server-side")
+type ServerSideApplyStrategy int
+
+const (
+	ServerSide ServerSideApplyStrategy = iota
+
+	ClientSide
+
+	ServerSideAuto
+)
+
+func GetServerSideApplyStrategy(cmd *cobra.Command) (ServerSideApplyStrategy, error) {
+	var serverSideFlag = GetFlagString(cmd, "server-side")
+	invalidOpt := fmt.Errorf(`invalid server-side value (%v). Must be one of true or false`, serverSideFlag)
+	if ServerSideAutoAlpha.IsEnabled() {
+		invalidOpt = fmt.Errorf(`invalid server-side value (%v). Must be one of true, false or auto`, serverSideFlag)
+	}
+
+	b, err := strconv.ParseBool(serverSideFlag)
+	// The flag is not a boolean - we only accept auto
+	if err != nil {
+		if serverSideFlag == "auto" {
```

Comment:
I'd suggest making the values of the ServerSidApplyStrategy constants match their expected flag values, so that you can use those here instead of hardcoding strings.